src/Symfony/Component/Filesystem/Path.php

```diff
@@ -0,0 +1,804 @@
+     * Returns whether a path is a base path of another path.
+     *
+     * Dot segments ("." and "..") are removed/collapsed and all slashes turned
+     * into forward slashes.
+     *
+     * ```php
+     * Path::isBasePath('/webmozart', '/webmozart/css');
+     * // => true
+     *
+     * Path::isBasePath('/webmozart', '/webmozart');
+     * // => true
+     *
+     * Path::isBasePath('/webmozart', '/webmozart/..');
+     * // => false
+     *
+     * Path::isBasePath('/webmozart', '/puli');
+     * // => false
+     * ```
+     */
+    public static function isBasePath(string $basePath, string $ofPath): bool
+    {
+        $basePath = self::canonicalize($basePath);
+        $ofPath = self::canonicalize($ofPath);
+
+        // Append slashes to prevent false positives when two paths have
+        // a common prefix, for example /base/foo and /base/foobar.
+        // Don't append a slash for the root "/", because then that root
+        // won't be discovered as common prefix ("//" is not a prefix of
+        // "/foobar/").
+        return 0 === strpos($ofPath.'/', rtrim($basePath, '/').'/');
+    }
+
+    /**
+     * Splits a canonical path into its root directory and the remainder.
+     *
+     * If the path has no root directory, an empty root directory will be
+     * returned.
+     *
+     * If the root directory is a Windows style partition, the resulting root
+     * will always contain a trailing slash.
+     *
+     * list ($root, $path) = Path::split("C:/webmozart")
+     * // => ["C:/", "webmozart"]
+     *
+     * list ($root, $path) = Path::split("C:")
+     * // => ["C:/", ""]
+     *
+     * @return string[] an array with the root directory and the remaining
+     *                  relative path
+     */
+    private static function split(string $path): array
+    {
+        if ('' === $path) {
+            return ['', ''];
+        }
+
+        // Remember scheme as part of the root, if any
+        if (false !== ($pos = strpos($path, '://'))) {
+            $root = substr($path, 0, $pos + 3);
+            $path = substr($path, $pos + 3);
+        } else {
+            $root = '';
+        }
+
+        $length = \strlen($path);
+
+        // Remove and remember root directory
+        if (0 === strpos($path, '/')) {
+            $root .= '/';
+            $path = $length > 1 ? substr($path, 1) : '';
+        } elseif ($length > 1 && ctype_alpha($path[0]) && ':' === $path[1]) {
+            if (2 === $length) {
+                // Windows special case: "C:"
+                $root .= $path.'/';
+                $path = '';
+            } elseif ('/' === $path[2]) {
+                // Windows normal case: "C:/"..
+                $root .= substr($path, 0, 3);
+                $path = $length > 3 ? substr($path, 3) : '';
+            }
+        }
+
+        return [$root, $path];
+    }
+
+    private static function toLower(string $str): string
+    {
+        return mb_strtolower($str, mb_detect_encoding($str));
```

Comment:
Should it fallback to `strtolower()` in case `mb_detect_encoding()` returns false? 
See e.g. https://github.com/symfony/symfony/blob/master/src/Symfony/Component/Console/Helper/Helper.php#L46

Reply:
👍